src/Circuit.java

```diff
@@ -0,0 +1,234 @@
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+
+/**
+ *
+ *
+ * @author Caden Hansen
+ */
+public final class Circuit {
+
+    /**
+     * Instance variables (fields)
+     */
+
+    /**
+     * arrayList to represent Circuit.
+     */
+    private ArrayList<Integer> circuit;
+
+    /**
+     * arrayList of arrayLists to allow for parallel circuits.
+     *
+     * not fully fleshed out rn, all the methods will likely need to be changed
+     * in someway to account for potential parallelCircuits but I think the
+     * utility would allow for more depth inside of the component. The ability
+     * to add comparators and other such circuit parts.
+     *
+     */
+    private ArrayList<ArrayList<Integer>> parallelCircuit;
+
+    /*
+     * location 0 voltage value.
+     */
+    private int voltage;
+
+    /**
+     * Default constructor.
+     */
+    private Circuit() {
+        //default voltage
+        this.voltage = 5;
+        this.circuit.add(0, this.voltage);
+
+    }
+
+    /**
+     *
+     * @param circuitList
+     * @param voltageValue
+     * @param parallelArray
+     *
+     *            Full argument constructor.
+     */
+    private Circuit(ArrayList<Integer> circuitList, int voltageValue,
+            ArrayList<ArrayList<Integer>> parallelArray) {
+        this.circuit = circuitList;
+        this.voltage = voltageValue;
+        this.parallelCircuit = parallelArray;
+        this.circuit.add(0, this.voltage);
+    }
+
+    /**
+     *
+     * @param quantity
+     *            number of wires to add to the list
+     */
+    public void addWire(int quantity) {
+
+        //starts with 1 because 0 value is always power source
```

Comment:
Excellent comments! These get right at the purpose of comments: to explain the **why**